src/mapreduce.jl

```diff
@@ -66,3 +70,23 @@ function Base.count(f, v::AbstractDiskArray)
         count(f, v[chunk...])
     end
 end
+
+for (_fname, init, acum) in ((:_sum, :zero, :+), (:_prod, :one, :*),
+                             (:_all, _->:true, :&), (:_any, _->:false, :|),
+                             (:_maximum, :typemin, :max), (:_minimum, :typemax, :min))
+	@eval Base.$_fname(a::AbstractDiskArray, dims)  = Base.$_fname(identity, a, dims)
+	@eval function Base.$_fname(f::Function, a::AbstractDiskArray{T,N}, dims) where {T,N}
+		_dims = typeof(dims)<:Tuple ? [dims...] : typeof(dims)<:Number ? [dims] : dims
+		out_dims = ntuple(Val(N)) do i
+			i in _dims ? 1 : size(a)[i]
+		end 
+		out = fill($init(Base.return_types(f, (T,))[1]), out_dims...)
```

Comment:
```suggestion
                T1 = Base.promote_op(f, T)
		out = fill($init(T1), out_dims...)
```

I think this is more likely to be stable? (untested)


Isn't this the syntax? `(T,)` wont be type stable.